test/system/src/test/java/io/pravega/test/system/AbstractFailoverTests.java

```diff
@@ -193,18 +199,34 @@ void performFailoverTest() throws InterruptedException {
                     } else {
                         log.debug("Read timeout");
                     }
-                } catch (ConnectionClosedException e) {
+                } catch (RetriesExhaustedException e) {
                     log.warn("Test exception in reading events: ", e);
                     continue;
-                } catch (ReinitializationRequiredException e) {
+                } catch (Throwable e) {
+                    if (e.getCause() instanceof RetriesExhaustedException) {
+                        log.warn("Test exception in reading events: ", e);
+                        continue;
```

Comment:
continue will not cause the test to error with this exception. This forces us to troll through the logs to find out if anything went wrong. If something goes wrong the test should immediately fail with the exception.